src/LeanCode.ContractsGenerator/Compilation/ContractTypes.cs

```diff
@@ -222,4 +230,25 @@ public bool IsReadOnlyDictionary(ITypeSymbol i)
                 || Dictionary.Contains(ns.ConstructUnboundGenericType())
             );
     }
+
+    public bool IsRecordEquatable(ITypeSymbol i)
+    {
+        return i is INamedTypeSymbol ns
+            && ns.IsGenericType
+            && ns.TypeArguments.FirstOrDefault() is INamedTypeSymbol namedType
+            && namedType.IsRecord
+            && Equatable.Contains(ns.ConstructUnboundGenericType());
+    }
+
+    /// <remarks>
```

Comment:
I don't think we should ever consider `ITypeSymbol` as an argument - this excludes the property, thus accepting `IPropertySymbol` in the first place makes sense.